src/blsToExecution.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Utils for converting a validator's withdrawal credentials from BLS12-381
+ * to an ETH1 address.
+ */
+import { ByteVectorType, ContainerType, UintNumberType, } from '@chainsafe/ssz';
+import { Client, Constants as SDKConstants } from 'gridplus-sdk'
+import { DOMAINS, NETWORKS } from './constants';
+import { ensureHexBuffer, buildSigningRoot } from './utils';
+
+/**
+ * Set a new ETH1 address for a validator's withdrawal credentials.
+ * This requires a signature.
+ * @param client - Instance of GridPlus SDK, which is already connected/paired to a target Lattice.
+ * @param path - Path of deposit/validator key. Array with up to five u32 indices representing BIP39 path.
+ * @param opts - Instance of `BlsToExecutionOpts` containing params to change the withdrawal credentials.
+ * @returns `SignedBlsToExecutionChange` object
+ */
+export async function generateObject(
+  client: Client,
+  path: number[],
+  opts: BlsToExecutionOpts,
+) : Promise<SignedBlsToExecutionChange> {
+  // Setup options/params
+  const {
+    eth1Addr,
+    validatorIdx,
+    networkInfo=NETWORKS.MAINNET_GENESIS,
+  } = opts;
+  if (!eth1Addr) {
+    throw new Error('No `eth1Addr` value found.');
+  } else if (validatorIdx === undefined || validatorIdx === null) {
+    throw new Error('No `validatorIdx` value found.');
+  }
+  // Convert eth1Addr to a Buffer
+  const eth1AddrBuf = ensureHexBuffer(eth1Addr);
```

Comment:
Should probably check that this is a 20 byte buffer here. I think it's implicitly checked later but an explicit check is warranted.

Reply:
[56356af](https://github.com/GridPlus/lattice-eth2-utils/pull/4/commits/56356afaabd3aef930dc605b776e7fcb3ce99431)